src/deploy/functions/tasks.ts

```diff
@@ -0,0 +1,279 @@
+      return operationResult;
+    } catch (err) {
+      params.errorHandler.record("error", fn.name, "create", err.message || "");
+    }
+  };
+}
+
+export function updateFunctionTask(
+  params: TaskParams,
+  fn: CloudFunctionTrigger,
+  onPoll?: (op: any) => any
+): () => Promise<CloudFunctionTrigger | void> {
+  return async () => {
+    utils.logBullet(
+      clc.bold.cyan("functions: ") +
+        "updating " +
+        getHumanFriendlyRuntimeName(params.runtime) +
+        " function " +
+        clc.bold(helper.getFunctionLabel(fn.name)) +
+        "..."
+    );
+    params.timer.startTimer(fn.name, "update");
+    const eventType = fn.eventTrigger ? fn.eventTrigger.eventType : "https";
+    try {
+      const updateRes = await cloudfunctions.updateFunction({
+        projectId: params.projectId,
+        region: helper.getRegion(fn.name),
+        eventType: eventType,
+        functionName: helper.getFunctionName(fn.name),
+        entryPoint: fn.entryPoint,
+        trigger: helper.getFunctionTrigger(fn),
+        labels: Object.assign({}, deploymentTool.labels(), fn.labels),
+        sourceUploadUrl: params.sourceUrl,
+        sourceToken: params.sourceToken,
+        runtime: params.runtime,
+        availableMemoryMb: fn.availableMemoryMb,
+        timeout: fn.timeout,
+        maxInstances: fn.maxInstances,
+        environmentVariables: fn.environmentVariables,
+        vpcConnector: fn.vpcConnector,
+        vpcConnectorEgressSettings: fn.vpcConnectorEgressSettings,
+        serviceAccountEmail: fn.serviceAccountEmail,
+      });
+      const pollerOptions: OperationPollerOptions = Object.assign(
+        {
+          pollerName: `update-${fn.name}`,
+          operationResourceName: updateRes.name,
+          onPoll,
+        },
+        defaultPollerOptions
+      );
+      const operationResult = await pollOperation<CloudFunctionTrigger>(pollerOptions);
+      params.timer.endTimer(fn.name);
+      helper.printSuccess(fn.name, "update");
+      return operationResult;
+    } catch (err) {
+      params.errorHandler.record("error", fn.name, "update", err.message || "");
+    }
+  };
+}
+
+export function deleteFunctionTask(params: TaskParams, fnName: string) {
+  return async () => {
+    utils.logBullet(
+      clc.bold.cyan("functions: ") +
+        "deleting function " +
+        clc.bold(helper.getFunctionLabel(fnName)) +
+        "..."
+    );
+    params.timer.startTimer(fnName, "delete");
+    try {
+      const deleteRes = await cloudfunctions.deleteFunction({
+        functionName: fnName,
+      });
+      const pollerOptions: OperationPollerOptions = Object.assign(
+        {
+          pollerName: `delete-${fnName}`,
+          operationResourceName: deleteRes.name,
+        },
+        defaultPollerOptions
+      );
+      const operationResult = await pollOperation<void>(pollerOptions);
+      params.timer.endTimer(fnName);
+      helper.printSuccess(fnName, "delete");
```

Comment:
If I'm reading this correctly, I think this will print fully qualified name (including project, region) vs other tasks print function name only. Is that okay?

Reply:
This is working as intended - printSuccess takes a fully qualified name. CloudFunctionTrigger.name is the fully qualified name, so the other tasks behave the same way.